python/paddle/fluid/tests/unittests/dist_transformer.py

```diff
@@ -440,9 +440,7 @@ def split_data(data, num_part):
 def test_context(train_progm, avg_cost, train_exe, dev_count, data_input_names,
                  sum_cost, token_num):
     # Context to do validation.
-    test_program = train_progm.clone()
-    with fluid.program_guard(test_program):
-        test_program = fluid.io.get_inference_program([avg_cost])
+    test_program = train_progm.clone(for_test=True)
```

Comment:
Does this work? I'm afraid that the train_program contains backward already. It should probably be called before Optimizer

Reply:
After move `test_program = train_progm.clone(for_test=True)` before Optimizer, `test_dist_transformer` runs OK.
```
[07:05:52][Step 1/1]         Start 401: test_dist_transformer
[07:10:11][Step 1/1] 424/428 Test #401: test_dist_transformer ...........................   Passed  259.23 sec
```
http://ci.paddlepaddle.org/viewLog.html?buildId=19035&tab=buildLog&buildTypeId=Paddle_PrCi&logTab=tree&filter=all&_focus=18687